File: SocialNetwork.WebUI/src/models/responses/Board.ts
import { Guid } from "@/utilities/guid";

export interface IBoard {
    id: Guid
    boardTypeId: Guid
    isArchived: boolean
    name: string
    description: string
    settings: IBoardSetting[]
}

export interface IBoards {
    boards: IBoard[]
}

export interface IBoardSetting {
    name: string
    value: string
}

File: SocialNetwork/SocialNetwork.Dal/Models/Board.cs
using System;
using System.Collections.Generic;

namespace SocialNetwork.Dal.Models
{
    public class Board
    {
        public Board()
        {
            BoardSettings = new HashSet<BoardSetting>();
        }

        public Guid Id { get; set; }
        public Guid BoardTypeId { get; set; }
        public DateTime CreateDateTime { get; set; }
        public bool IsArchived { get; set; }

        public string Name { get; set; }
        public string Description { get; set; }

        public virtual BoardType BoardType { get; set; }
        public virtual ICollection<BoardSetting> BoardSettings { get; set; }
    }
}
